awx/main/analytics/core.py

```diff
@@ -51,40 +74,26 @@ def projects_by_scm_type():
     def decorate(f):
         f.__awx_analytics_key__ = key
         f.__awx_analytics_version__ = version
+        f.__awx_analytics_description__ = description
+        f.__awx_analytics_type__ = format
+        f.__awx_expensive__ = expensive
         return f
 
     return decorate
 
 
-def table_version(file_name, version):
-
-    global manifest
-    manifest[file_name] = version
-
-    def decorate(f):
-        return f
-
-    return decorate
-
-
-def gather(dest=None, module=None, collection_type='scheduled'):
+def gather(dest=None, module=None, subset = None, since = None, until = now(), collection_type='scheduled'):
     """
     Gather all defined metrics and write them as JSON files in a .tgz
 
     :param dest:    the (optional) absolute path to write a compressed tarball
-    :pararm module: the module to search for registered analytic collector
+    :param module: the module to search for registered analytic collector
                     functions; defaults to awx.main.analytics.collectors
     """
 
-    run_now = now()
-    state = TowerAnalyticsState.get_solo()
-    last_run = state.last_run
-    logger.debug("Last analytics run was: {}".format(last_run))
+    last_run = since or settings.AUTOMATION_ANALYTICS_LAST_GATHER or (now() - timedelta(weeks=4))
```

Comment:
the `since` and `until` will be always set now correct? So we should just do

```
last_run = since
```

So we make sure the time window is set only on 1 place.

Reply:
ideally we'd change the method to:
```
def gather(since, until, dest=None, module=None, subset = None, collection_type='scheduled')
```

Since the `since` and `until` must be always sent, as the fixed time window

Reply:
awx/main/mangement/commands/gather_analytics.py (manual gathering) doesn't set both timestamps when calling `gather()`. It could, and then we could change this.

Reply:
I've updated the management command to *allow* for setting a since and until, but don't want to require it. So I'll leave this like this.

Reply:
so we load this in tasks already as
```
last_gather = Setting.objects.filter(key='AUTOMATION_ANALYTICS_LAST_GATHER').first()
```

So Ideally we don't do this on multiple places.

To avoid bugs, there should be 1 place setting the timeframe, then all collections would use the same timeframe.